### ISSUE TITLE
Please improve error message when binding null values to a TextField

### ISSUE THREAD

**Issue by mvysny** [vaadin/framework#11109]
I am implementing a simple CRUD dialog in Vaadin 8.5.1. I create a new JPA bean and pass it to Binder.readBean, however it fails with:
```
java.lang.NullPointerException: value cannot be null
	at java.util.Objects.requireNonNull(Objects.java:228)
	at com.vaadin.ui.AbstractTextField.setValue(AbstractTextField.java:96)
	at com.vaadin.ui.AbstractTextField.setValue(AbstractTextField.java:46)
	at com.vaadin.data.Binder$BindingImpl.initFieldValue(Binder.java:890)
	at com.vaadin.data.Binder$BindingImpl.access$100(Binder.java:763)
	at com.vaadin.data.Binder.lambda$readBean$11(Binder.java:1380)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at com.vaadin.data.Binder.readBean(Binder.java:1380)
```
From the message it is impossible to tell which field is errorneous.

I'm thinking of having a `HasValue.supportsNullValues()` method (for which `AbstractTextField` would return false); Binder could then check whether he's trying to set a `null` value to a `HasValue` that doesn't support null values, and if yes, he can throw a NPE with an informative message including the field name.

**Comment by Legioth** [comment]
An alternative implementation of the same idea for Flow: https://github.com/vaadin/flow/pull/6757

Rather than introducing a new method like `supportsNullValues()`, I'm instead checking for a non-null empty value as a hint about whether the field supports null values. This will cover at least cover the main known case with `TextField`.

**Comment by TatuLund** [comment]
Closed by #11758
